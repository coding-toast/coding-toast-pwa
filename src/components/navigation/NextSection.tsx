import * as React from 'react'
import { observer } from 'mobx-react-lite'
import { NextSectionStoreContext } from '../../stores/NextSectionStore'
import NextSectionImage from '../../images/next-section.png'
import NextSectionStyles from './NextSection.module.scss'

const NextSection: React.FC = observer(() => {
  const nextSectionStore = React.useContext(NextSectionStoreContext)
  const executeScroll = () => {
    window.scrollTo({
      top: nextSectionStore.ref.current?.offsetTop,
      behavior: 'smooth'
    })
  }
  return (
    <button type="button" onClick={executeScroll} onKeyPress={executeScroll} tabIndex={-1} className={NextSectionStyles.chevronButton}>
      <img src={NextSectionImage} alt="Next Section" />
    </button>
  )
})

export default NextSection
