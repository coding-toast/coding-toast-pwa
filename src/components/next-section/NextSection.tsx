import * as React from 'react';
import NextSectionStyles from './NextSection.module.scss';

const NextSection: React.FC = () => {
  /*
  TODO:
  const nextSectionStore = React.useContext(NextSectionStoreContext)
  const offset: any = nextSectionStore.ref.current?.offsetParent

  const executeScroll = () => {
    window.scrollTo({
      top: offset?.offsetTop - 77,
      behavior: 'smooth'
    })
  }
  */

  return (
    <button type='button' tabIndex={0} className={NextSectionStyles.chevronButton}>
      <img src='/next-section.png' alt='Next Section' />
    </button>
  );
};

export default NextSection;
