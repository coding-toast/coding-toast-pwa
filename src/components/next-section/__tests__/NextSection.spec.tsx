import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NextSection from '../NextSection';

describe('<NextSection />', () => {
  it('renders correctly', () => {
    const component = shallow(<NextSection />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
