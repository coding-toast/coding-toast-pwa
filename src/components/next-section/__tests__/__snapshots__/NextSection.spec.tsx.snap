// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NextSection /> renders correctly 1`] = `
<button
  tabIndex={0}
  type="button"
>
  <img
    alt="Next Section"
    src="/next-section.png"
  />
</button>
`;
